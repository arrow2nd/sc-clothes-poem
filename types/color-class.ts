export type ColorClass = {
  idolName: string
  className: string
}
