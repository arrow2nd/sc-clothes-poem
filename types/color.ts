export type Color = {
  idolName: string
  className: string
  hex: string
}
