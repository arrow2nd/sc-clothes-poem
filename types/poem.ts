export type Poem = {
  id: string;
  idolName: string;
  clothesTitle: string;
  clothesName: string;
  text: string;
};
