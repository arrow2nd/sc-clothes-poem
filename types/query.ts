export type Query = {
  type?: "poem" | "idol" | "clothe";
  q?: string;
};
