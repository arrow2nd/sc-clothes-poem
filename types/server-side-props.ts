export type ServerSideProps = {
  ogpImageUrl: string;
  poemText: string;
};
