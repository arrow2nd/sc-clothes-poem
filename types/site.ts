export type Site = {
  title: string;
  desc: string;
  url: string;
};
