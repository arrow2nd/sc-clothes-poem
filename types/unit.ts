export type Unit = {
  name: string;
  members: string[];
};
