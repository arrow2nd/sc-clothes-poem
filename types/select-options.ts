import { Unit } from "./unit";

export type SelectOptions = {
  units: Unit[];
  clothes: string[];
};
