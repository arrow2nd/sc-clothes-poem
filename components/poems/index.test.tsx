import { render } from "@testing-library/react";

import { Poem } from "types/poem";

import Poems from "./index";

describe("Poems", () => {
  const samplePoems: Poem[] = [
    {
      id: "FashionableSummer_YukokuKiriko",
      idolName: "幽谷霧子",
      clothesTitle: "ファッショナブルサマー",
      clothesName: "ファッショナブルサマー",
      text: "サマーダイブ！波がさらった五線譜のように"
    },
    {
      id: "SunsetSkyPassage_YukokuKiriko",
      idolName: "幽谷霧子",
      clothesTitle: "サンセットスカイパッセージ",
      clothesName: "サンセットスカイパッセージ",
      text: "スタッカート。まなうらにほら、赤橙黄緑霧藍紫"
    }
  ];

  test("ポエムの一覧表示が正しいか", () => {
    const { container } = render(<Poems items={samplePoems} />);
    expect(container).toMatchSnapshot();
  });

  test("該当ポエムがない場合の表示が正しいか", () => {
    const { container } = render(<Poems items={[]} />);
    expect(container).toMatchSnapshot();
  });
});
