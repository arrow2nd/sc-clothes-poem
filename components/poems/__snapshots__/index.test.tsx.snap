// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poems ポエムの一覧表示が正しいか 1`] = `
<div>
  <div
    class="mb-16 font-kiwi"
  >
    <div
      class="flex flex-wrap justify-center"
    >
      <div
        class="relative flex items-center w-96 h-60 m-2 p-8 border-2 border-main rounded-md bg-white text-main"
      >
        <div>
          <div
            class="mb-4 text-lg md:text-xl"
          >
            <p>
              サマーダイブ！
            </p>
            <p>
              波がさらった五線譜のように
            </p>
          </div>
          <div
            class="mb-4 py-0.5 w-10 rounded-full border border-gray-100"
            style="background-color: rgb(217, 242, 255);"
          />
          <div
            class=""
          >
            <p
              class="mb-1 text-sm md:text-base"
            >
              ファッショナブルサマー
            </p>
            <p
              class="text-xs md:text-sm"
            >
              幽谷霧子
            </p>
          </div>
        </div>
        <div
          class="absolute right-0 bottom-0 p-8 flex flex-row space-x-4 text-main"
        >
          <a
            class="text-xl hover:text-black transition-colors"
            href="https://twitter.com/intent/tweet?text=%E3%82%B5%E3%83%9E%E3%83%BC%E3%83%80%E3%82%A4%E3%83%96%EF%BC%81%0A%E6%B3%A2%E3%81%8C%E3%81%95%E3%82%89%E3%81%A3%E3%81%9F%E4%BA%94%E7%B7%9A%E8%AD%9C%E3%81%AE%E3%82%88%E3%81%86%E3%81%AB%0A%23%E3%82%B7%E3%83%A3%E3%83%8B%E3%83%9E%E3%82%B9%20%23%E3%83%95%E3%82%A1%E3%83%83%E3%82%B7%E3%83%A7%E3%83%8A%E3%83%96%E3%83%AB%E3%82%B5%E3%83%9E%E3%83%BC%20%23%E5%B9%BD%E8%B0%B7%E9%9C%A7%E5%AD%90%0Ahttps%3A%2F%2Fshiny-poems.vercel.app%3Fid%3DFashionableSummer_YukokuKiriko"
            rel="noopener noreferrer"
            target="_blank"
            title="ツイートする"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
              />
            </svg>
          </a>
          <button
            class="text-xl hover:text-black transition-colors focus:outline-none"
            title="コピーする"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="13"
                rx="2"
                ry="2"
                width="13"
                x="9"
                y="9"
              />
              <path
                d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="relative flex items-center w-96 h-60 m-2 p-8 border-2 border-main rounded-md bg-white text-main"
      >
        <div>
          <div
            class="mb-4 text-lg md:text-xl"
          >
            <p>
              スタッカート。
            </p>
            <p>
              まなうらにほら、赤橙黄緑霧藍紫
            </p>
          </div>
          <div
            class="mb-4 py-0.5 w-10 rounded-full border border-gray-100"
            style="background-color: rgb(217, 242, 255);"
          />
          <div
            class=""
          >
            <p
              class="mb-1 text-sm md:text-base"
            >
              サンセットスカイパッセージ
            </p>
            <p
              class="text-xs md:text-sm"
            >
              幽谷霧子
            </p>
          </div>
        </div>
        <div
          class="absolute right-0 bottom-0 p-8 flex flex-row space-x-4 text-main"
        >
          <a
            class="text-xl hover:text-black transition-colors"
            href="https://twitter.com/intent/tweet?text=%E3%82%B9%E3%82%BF%E3%83%83%E3%82%AB%E3%83%BC%E3%83%88%E3%80%82%0A%E3%81%BE%E3%81%AA%E3%81%86%E3%82%89%E3%81%AB%E3%81%BB%E3%82%89%E3%80%81%E8%B5%A4%E6%A9%99%E9%BB%84%E7%B7%91%E9%9C%A7%E8%97%8D%E7%B4%AB%0A%23%E3%82%B7%E3%83%A3%E3%83%8B%E3%83%9E%E3%82%B9%20%23%E3%82%B5%E3%83%B3%E3%82%BB%E3%83%83%E3%83%88%E3%82%B9%E3%82%AB%E3%82%A4%E3%83%91%E3%83%83%E3%82%BB%E3%83%BC%E3%82%B8%20%23%E5%B9%BD%E8%B0%B7%E9%9C%A7%E5%AD%90%0Ahttps%3A%2F%2Fshiny-poems.vercel.app%3Fid%3DSunsetSkyPassage_YukokuKiriko"
            rel="noopener noreferrer"
            target="_blank"
            title="ツイートする"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
              />
            </svg>
          </a>
          <button
            class="text-xl hover:text-black transition-colors focus:outline-none"
            title="コピーする"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="13"
                rx="2"
                ry="2"
                width="13"
                x="9"
                y="9"
              />
              <path
                d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Poems 該当ポエムがない場合の表示が正しいか 1`] = `
<div>
  <div
    class="mb-16 font-kiwi"
  >
    <div
      class="flex flex-row justify-center items-center py-8 text-main text-lg md:text-xl"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
        />
        <path
          d="M4 10.5a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5zm5 0a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5z"
        />
      </svg>
      <span
        class="ml-2"
      >
        ポエムが見つかりません…
      </span>
    </div>
  </div>
</div>
`;
